src/server/config.rs

```diff
@@ -405,6 +403,31 @@ impl Config {
             GrpcCompressionType::Gzip => CompressionAlgorithms::GRPC_COMPRESS_GZIP,
         }
     }
+
+    pub fn end_point_request_max_handle_duration(&self) -> ReadableDuration {
+        if let Some(end_point_request_max_handle_duration) =
+            self.end_point_request_max_handle_duration
+        {
+            return end_point_request_max_handle_duration;
+        }
+        ReadableDuration::secs(DEFAULT_ENDPOINT_REQUEST_MAX_HANDLE_SECS)
+    }
+
+    pub fn optimize_for(&mut self, region_size: ReadableSize) {
+        // It turns out for 256MB region size, 60s is typically enough.
+        const THRESHOLD_SIZE: ReadableSize = ReadableSize::mb(256);
+        if region_size.0 < THRESHOLD_SIZE.0 {
+            self.end_point_request_max_handle_duration
+                .get_or_insert(ReadableDuration::secs(
+                    DEFAULT_ENDPOINT_REQUEST_MAX_HANDLE_SECS,
+                ));
+        } else {
+            self.end_point_request_max_handle_duration
+                .get_or_insert(ReadableDuration::secs(
+                    region_size.0 / THRESHOLD_SIZE.0 * DEFAULT_ENDPOINT_REQUEST_MAX_HANDLE_SECS,
```

Comment:
It means the timeout is 40 minutes for a single cop req when region size is 10Gb. Is it too large?

Reply:
make it up to 30 min instead.